sad_comparison_functions.py

```diff
@@ -92,4 +154,30 @@ def get_obs_pred_multi_dists(dat_dir, file_name, dist_name, cutoff = 9):
                 results['f2'] = pred_dist_site
                 out.writerows(results)
     out_write.close()
-            
\ No newline at end of file
+            
+def sim_stats(ab, dist_name, Nsim, test_stat):
+    """Obtain Nsim abundance lists from the proposed distribution 
+    
+    and compare their fit with the empirical data (Connolly et al. 2009).
+    Inputs:
+    ab - list of empirical abundaces
+    dist_name - name of the distribution under examination
+    Nsim - number of simulated abundance lists
+    test_stat - can be either log-likelihood ('loglik'), Kolmogorov-Smirnov statistic ('ks'), or R^2 ('r2')
+    Output: 
+    A list where the first element is the stat value for the empirical SAD and the remaining values are stats for the simulated SADs.
+    
+    """
+    pars = get_par_multi_dists(ab, dist_name)
+    if test_stat == 'loglik': test_func = get_loglik_multi_dists
+    elif test_stat == 'ks': test_func = get_ks_multi_dists
+    elif test_stat == 'r2': 
+        def test_func(ab, dist_name, pars):
+            pred = get_pred_multi_dists(len(ab), dist_name, pars)
+            r2 = macroecotools.obs_pred_rsquare(sorted(ab, reverse = True), pred)
+            return r2
+    out_list = [test_func(ab, dist_name, pars)]
+    for i in range(Nsim):
+        sim_sad = get_sample_multi_dists(len(ab), dist_name, pars)
+        out_list.append(test_func(sim_sad, dist_name, pars))
+    return out_list
```

Comment:
Yeah, I was actually debating about that but decided to go the simplier (by
eye) route. But changing the output to tuple would be an easy fix.

On Thu, Feb 12, 2015 at 10:58 AM, Ethan White <notifications@github.com>
wrote:

> In sad_comparison_functions.py
> <https://github.com/weecology/sad-comparison/pull/112#discussion_r24591265>
> :
>
> > +    A list where the first element is the stat value for the empirical SAD and the remaining values are stats for the simulated SADs.
> > +
> > +    """
> > +    pars = get_par_multi_dists(ab, dist_name)
> > +    if test_stat == 'loglik': test_func = get_loglik_multi_dists
> > +    elif test_stat == 'ks': test_func = get_ks_multi_dists
> > +    elif test_stat == 'r2':
> > +        def test_func(ab, dist_name, pars):
> > +            pred = get_pred_multi_dists(len(ab), dist_name, pars)
> > +            r2 = macroecotools.obs_pred_rsquare(sorted(ab, reverse = True), pred)
> > +            return r2
> > +    out_list = [test_func(ab, dist_name, pars)]
> > +    for i in range(Nsim):
> > +        sim_sad = get_sample_multi_dists(len(ab), dist_name, pars)
> > +        out_list.append(test_func(sim_sad, dist_name, pars))
> > +    return out_list
>
> I think outputing the value for the empirical data and those for the sims
> all in a single list has the potential to be confusing. I'd recommend
> returning a tuple that contains the value for the empirical data as the
> first value and the list of values for the sims as the second value. Does
> that make sense?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/weecology/sad-comparison/pull/112/files#r24591265>.
>